File: java/src/main/java/com/yofei/shortlink/dao/entity/LinkEntity.java
package com.yofei.shortlink.dao.entity;

import com.yofei.shortlink.common.Persistable;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "link")
public class LinkEntity extends Persistable {

    @Column(nullable = false)
    private String md5;

    @Column(nullable = false)
    private String url;

}

File: fullstack/src/entities/link.entity.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';
import { BaseEntity } from './base.entity';

@Entity({ name: 'sl_link' })
export class LinkEntity extends BaseEntity {
  @PrimaryColumn({ type: 'varchar', length: 8 })
  id: string;

  @Column({ type: 'text' })
  link: string;
}
